hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java

```diff
@@ -814,6 +814,11 @@ public void leftRight_substr() {
 				.register();
 	}
 
+	/**
```

Comment:
Since this is only used for Derby, we should move it to the DerbyDialect.  It is hardly "common".

Now, yes, it is also used by the DerbyLegacyDialect, but imo we should either (1) drop that Dialect when we move DerbyDialect to hibernate-community-dialects or (2) just have DerbyLegacyDialect use the one from DerbyDialect.  Is there really a reason to keep DerbyLegacyDialect around?